### PR TITLE
groff: Add site.tmac to fix man page coloring

### PR DESCRIPTION
**Pull request by aneeshusa**
This enables groff to correctly respond to the LESS_TERMCAP_*
environment variables, allowing colorized man pages.

Patch is taken from Arch Linux;
see https://bugs.archlinux.org/task/33760.

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/groff/site.tmac [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+.  \" Character translations for non-keyboard
+.  \" characters - to make them searchable
+.  if '\*[.T]'utf8' \{\
+.    char \- \N'45'
+.    char - \N'45'
+.    char ' \N'39'
+.    char \' \N'39'
+.  \}
+.
+.  \" Shut off SGR by default (groff colors)
```
Comment (0xABAB) [inline pkgs/tools/text/groff/site.tmac]: `SGR` -> `SGR (a.k.a.  ANSI color escapes)`

File: pkgs/tools/text/groff/site.tmac [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+.  \" Require GROFF_SGR envvar defined to turn it on
```
Comment (0xABAB) [inline pkgs/tools/text/groff/site.tmac]: Minor, but I'd just write out "environment variable" here.

File: pkgs/tools/text/groff/default.nix [hunk 3]
```diff
@@ -59,6 +59,9 @@ stdenv.mkDerivation rec {
   postInstall = ''
     rm $doc/share/doc/groff/examples/hdtbl/*color*ps
     find $doc/share/doc/groff/ -type f -print0 | xargs -0 sed -i -e 's/%%CreationDate: .*//'
+    for f in 'man.local' 'mdoc.local'; do
```
Comment (0xABAB) [inline pkgs/tools/text/groff/default.nix]: You don't need to quote these constants.

File: pkgs/tools/text/groff/default.nix [hunk 4]
```diff
@@ -59,6 +59,9 @@ stdenv.mkDerivation rec {
...
+        cat '${./site.tmac}' >>"$out/share/groff/site-tmac/$f"
```
Comment (0xABAB) [inline pkgs/tools/text/groff/default.nix]: Are you sure this does anything useful? (The single quotes seem to be off, turning cat's argument into a constant.)

File: pkgs/tools/text/groff/default.nix [hunk 5]
```diff
@@ -59,6 +59,9 @@ stdenv.mkDerivation rec {
   postInstall = ''
     rm $doc/share/doc/groff/examples/hdtbl/*color*ps
```
Comment (0xABAB) [inline pkgs/tools/text/groff/default.nix]: `"$doc"`

File: pkgs/tools/text/groff/default.nix (hunk 4 above)
Comment (Mic92) [inline pkgs/tools/text/groff/default.nix]: '${./site.tmac}' expands to '/nix/store/<hash>-site.tmac' before the build script is executed.

**Comment by aneeshusa** [comment]
@Mic92 I didn't realize this was a mass-rebuild, I switched the base to staging.

@0xABAB, since I'm pulling it from an upstream I'd prefer to avoid cosmetic changes to the `site.tmac` file.

File: pkgs/tools/text/groff/default.nix (hunk 5 above)
Comment (aneeshusa) [inline pkgs/tools/text/groff/default.nix]: This is unrelated to the change, so I'm going to leave it untouched in this PR.

Comment (vcunat) [inline pkgs/tools/text/groff/default.nix]: Done in 1db1ba904.

**Comment by aneeshusa** [comment]
@vcunat looks like you merged this into staging already and addressed the other nit, so closing the PR. Thanks!

**Comment by vcunat** [comment]
@aneeshusa: this causes a blocker problem &ndash; `nixos-rebuild --help | grep SYNOPSIS` won't work anymore, breaking `tests.misc.*`.  I don't know why that would happen, as it shows OK when I pipe that into `less` or `cat`...

**Comment by vcunat** [comment]
I unblocked this one now, but it would still be nice to understand if it can cause some *real* problems.
